### PR TITLE
Issue/lrs statement counts

### PR DESCRIPTION
**Pull request by ht2**
Unifies the method for getting statement counts using the native count method. Addresses issue #760

### CODE REVIEW

File: app/controllers/SiteController.php [hunk 1]
```diff
@@ -120,11 +120,13 @@ public function getGraphData(){
...
     $lrss = $this->lrs->index($opts);
-    $counts = $this->lrs->getCounts();
+    $lrs_repo = $this->lrs;
 
+    $collection = \DB::getCollection('statements');
 
-    return Response::json(array_map(function ($lrs) use ($counts) {
-      $lrs->statement_total = isset($counts[$lrs->_id]) ? $counts[$lrs->_id] : 0;
+    return Response::json(array_map(function ($lrs) use ($lrs_repo) {      
+      $query = ['lrs_id'=>new \MongoId($lrs->_id)];
```
Comment (ryansmith94) [inline app/controllers/SiteController.php]: I think line 128 can be removed. I can't see where it's used.

File: app/locker/data/dashboards/BaseDashboard.php [hunk 2]
```diff
@@ -34,17 +37,8 @@ public function getGraphData(\DateTime $startDate = null, \DateTime $endDate = n
...
-
-    $query = [];
-    
-    //Limit by LRS
-    if( $this->has_lrs ){
-      $query['lrs_id'] = new \MongoId($this->lrs);
-    }
-
-    return $collection->count($query);
+    $lrs_id = $this->has_lrs ? $this->lrs : null;
```
Comment (ryansmith94) [inline app/locker/data/dashboards/BaseDashboard.php]: If there's a method that *needs* to be called externally before this method to set `$this->has_lrs` can that be commented somewhere?

Comment (ht2) [inline app/locker/data/dashboards/BaseDashboard.php]: The constructor sets it as a boolean depending on whether an $lrs (id) was passed in as an argument

File: app/controllers/SiteController.php (hunk 1 above)
Comment (ht2) [inline app/controllers/SiteController.php]: Yea you're right, I moved the count to the repo method and the query didn't get removed

File: app/locker/data/dashboards/BaseDashboard.php (hunk 2 above)
Comment (ht2) [inline app/locker/data/dashboards/BaseDashboard.php]: Sorry that wasn't quite correct, it is set in the constructor, but explicitly depending on the type of dashboard that extends the BaseDashboard (which should really be abstract). E.g the LrsDashboard declares itself as having an LRS...
https://github.com/LearningLocker/learninglocker/blob/issue/lrs-statement-counts/app/locker/data/dashboards/LrsDashboard.php#L11

Comment (ryansmith94) [inline app/locker/data/dashboards/BaseDashboard.php]: Ahhhh I get it. Probably could be static then or $this->lrs could be set to null and remove the variable.

**Pull request merged**